core/templates/dev/head/exploration_editor/EditorServices.js

```diff
@@ -1952,3 +2012,302 @@ oppia.factory('explorationWarningsService', [
+
+  // Detects whether an object of the type 'answer_group' or 'default_outcome'
+  // has been added, edited or deleted.
+  // Returns - 'addded', 'edited' or 'deleted' accordingly.
+  var getRelativeChangeToGroups = function(changeObject) {
+    var newValue = changeObject.new_value;
+    var oldValue = changeObject.old_value;
+    var result = '';
+
+    if (angular.isArray(newValue) && angular.isArray(oldValue)) {
+      result = (newValue.length > oldValue.length) ?
+        'added' : (newValue.length === oldValue.length) ?
+        'edited' : 'deleted';
+    } else {
+      if (!utilsService.isEmpty(oldValue)) {
+        if (!utilsService.isEmpty(newValue)) {
+          result = 'edited';
+        } else {
+          result = 'deleted';
+        }
+      } else if (!utilsService.isEmpty(newValue)) {
+        result = 'added';
+      }
+    }
+    return result;
+  };
+
+  var makeHumanReadable = function(lostChanges) {
+    var outerHtml = angular.element('<ul></ul>');
+    var stateWiseEditsMapping = {};
+    // The variable stateWiseEditsMapping stores the edits grouped by state.
+    // For instance, you made the following edits:
+    // 1. Changed content to 'Welcome' instead of '' in 'First Card'.
+    // 2. Added an interaction in this state.
+    // 2. Added a new state 'End'.
+    // 3. Ended Exporation from state 'End'.
+    // stateWiseEditsMapping will look something like this:
+    // - 'First Card': [
+    //   - 'Edited Content: Welcome!',:
+    //   - 'Added Interaction: Continue',
+    //   - 'Added interaction customizations']
+    // - 'End': ['Ended exploration']
+
+    lostChanges.forEach(function(lostChange) {
+      var lostChangeEl = angular.element('<li></li>');
+      var lostChangeValue;
+
+      switch (lostChange.cmd) {
+        case CMD_ADD_STATE:
+          outerHtml.append(
+            angular.element('<li></li>').html(
+              'Added state: ' + lostChange.state_name));
+          break;
+        case CMD_RENAME_STATE:
+          outerHtml.append(
+            angular.element('<li></li>').html(
+              'Renamed state: ' + lostChange.old_state_name + ' to ' +
+                lostChange.new_state_name));
+          break;
+        case CMD_DELETE_STATE:
+          outerHtml.append(
+            angular.element('<li></li>').html(
+              'Deleted state: ' + lostChange.state_name));
+          break;
+        case CMD_EDIT_STATE_PROPERTY:
+          var newValue = getStatePropertyValue(lostChange.new_value);
+          var oldValue = getStatePropertyValue(lostChange.old_value);
+          var stateName = lostChange.state_name;
+          if (!stateWiseEditsMapping[stateName]) {
+            stateWiseEditsMapping[stateName] = [];
+          }
+
+          switch (lostChange.property_name) {
+            case 'content':
+              if (newValue !== null) {
+                stateWiseEditsMapping[stateName].push(
+                  angular.element('<div></div>').html('Edited content: ' +
```

Comment:
Our style for this would be 
angular.element('<div></div>').html(
  'Edited content: ' + newValue.value));